src/journal-remote/journal-remote.c

```diff
@@ -777,21 +789,43 @@ static int setup_microhttpd_socket(RemoteServer *s,
         return setup_microhttpd_server(s, fd, key, cert, trust);
 }
 
+static int null_timer_event_handler(sd_event_source *timer_event, uint64_t usec, void *userdata)
+{
+        return dispatch_http_event(timer_event, 0, 0, userdata);
+}
+
 static int dispatch_http_event(sd_event_source *event,
                                int fd,
                                uint32_t revents,
                                void *userdata) {
         MHDDaemonWrapper *d = userdata;
         int r;
+        fd_set rd, w, e;
+        MHD_UNSIGNED_LONG_LONG timeout = ULONG_LONG_MAX;
 
         assert(d);
 
-        r = MHD_run(d->daemon);
+        r = MHD_get_fdset(d->daemon, &rd, &w, &e, 0);
+        if (r == MHD_NO) {
+                log_error_errno(r, "MHD_get_fdset failed: %m");
```

Comment:
log_error_errno() expects a proper `errno`-style error code as first argument, but "r" in this case is a minihttpd-style error code. Since we have no `errno`-style error code here, please use plain `log_error()` and drop the `r` argument.